### PR TITLE
Job workers: timeouts & resumable

### PR DESCRIPTION
**Pull request by DrDet**
relates to `KPHP-1257`

### CODE REVIEW

File: runtime/job-workers/job-interface.h [hunk 1]
```diff
@@ -45,8 +50,51 @@ struct C$KphpJobWorkerResponse : job_workers::SendingInstanceBase {
...
+class_instance<C$KphpJobWorkerResponseError> f$KphpJobWorkerResponseError$$__construct(class_instance<C$KphpJobWorkerResponseError> const &v$this) noexcept;
+string f$KphpJobWorkerResponseError$$getError(class_instance<C$KphpJobWorkerResponseError> const &v$this) noexcept;
+int64_t f$KphpJobWorkerResponseError$$getErrorCode(class_instance<C$KphpJobWorkerResponseError> const &v$this) noexcept;
+
 bool f$is_kphp_job_workers_enabled() noexcept;
 
 void global_init_job_workers_lib() noexcept;
 
-void process_job_worker_answer_event(job_workers::JobSharedMessage *job_result) noexcept;
+extern int job_timeout_wakeup_id;
```
Comment (AlexK0) [inline runtime/job-workers/job-interface.h:96]: Let's make it a private variable in the anonymous namespace and add getter function
get_job_timeout_wakeup_id();

File: runtime/job-workers/job-interface.cpp [hunk 2]
```diff
@@ -2,25 +2,62 @@
...
 
 #include "runtime/job-workers/job-interface.h"
 
 bool f$is_kphp_job_workers_enabled() noexcept {
   return vk::singleton<job_workers::JobWorkersContext>::get().job_workers_num > 0;
 }
 
+int job_timeout_wakeup_id{-1};
+
+static void process_job_timeout(kphp_event_timer *timer) {
```
Comment (AlexK0) [inline runtime/job-workers/job-interface.cpp:21]: please, anonymous namespace

File: runtime/job-workers/client-functions.cpp [hunk 3]
```diff
@@ -2,26 +2,92 @@
...
     php_warning("Can't send job: job workers disabled");
     return -1;
   }
+  if (request.is_null()) {
+    php_warning("Can't send job: the request shouldn't be null");
+    return -1;
+  }
+  if (timeout < 0) {
+    timeout = DEFAULT_SCRIPT_TIMEOUT;
+  } else if (timeout < 1 || timeout > MAX_SCRIPT_TIMEOUT) {
```
Comment (AlexK0) [inline runtime/job-workers/client-functions.cpp:87]: I guess 1 second is too much for minimum.. 
May be 1 ms is more reasonable?

File: server/job-workers/job-worker-server.cpp [hunk 4]
```diff
@@ -206,4 +208,35 @@ const char *JobWorkerServer::send_job_reply(JobSharedMessage *job_response) noex
   return nullptr;
 }
 
+void JobWorkerServer::try_store_job_response_error(const char *error_msg, int error_code) {
+  if (reply_was_sent) {
```
[the remark this answers is not in the document]
Comment (DrDet) [reply server/job-workers/job-worker-server.cpp:213]: added assertion that `running_job != nullptr` in this function

File: runtime/job-workers/job-interface.h (hunk 1 above)
Comment (DrDet) [reply runtime/job-workers/job-interface.h]: done

File: runtime/job-workers/job-interface.cpp (hunk 2 above)
Comment (DrDet) [reply runtime/job-workers/job-interface.cpp]: done

File: runtime/job-workers/client-functions.cpp (hunk 3 above)
Comment (DrDet) [reply runtime/job-workers/client-functions.cpp:87]: If script execution timeout < 0.2 sec +- eps, script will be terminated before the start:

https://github.com/VKCOM/kphp/blob/80d895840cfd88194964f182aa8485d178974c78/server/php-worker.cpp#L161

**Review by AlexK0 — DISMISSED** [review]

**Review by AlexK0 — APPROVED** [review]

**Review by unserialize — APPROVED** [review]

**Comment by unserialize** [comment]
Squash into a single commit, please

**Pull request merged**